cacher.go

```diff
@@ -80,10 +80,12 @@ func (c *Cacher) addOrReuseLayer(cache Cache, layer bpLayer, previousSHA string)
 	}
 
 	if sha == previousSHA {
-		c.Out.Printf("Reusing layer '%s' with SHA %s\n", layer.Identifier(), sha)
+		c.Logger.Infof("Reusing layer '%s'", layer.Identifier())
+		c.Logger.Debugf("  - with SHA %s\n", sha)
```

Comment:
This seems a little odd to me to be honest. Log messages should be able to stand on their own. The fact that there is no new line character in the `Infof` call might not work as you'd expect based on some of the opinions we've made in certain logger implementations. I believe we add a new line character per log message if not present when writing out.

IMO, a better approach would be two log messages.

```go
c.Logger.Infof("Reusing layer '%s'\n'", layer.Identifier())
c.Logger.Debugf("Layer '%s' SHA: %s\n", layer.Identifier(), sha)
```